### PR TITLE
chore: update static-site-generator-webpack-plugin

### PR DESCRIPTION
**Pull request by slorber**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Upgrade the dependency `@slorber/static-site-generator-webpack-plugin` after some cleanup and a few PRs merged

New `concurrency` option added with env variable. Not documented for now, looking for feedback before.

## Test Plan

site builds fine, not slower than before

### Test links

Deploy preview: https://deploy-preview-_____--docusaurus-2.netlify.app/

## Related issues/PRs

https://github.com/slorber/static-site-generator-webpack-plugin/pull/2

https://github.com/slorber/static-site-generator-webpack-plugin/pull/3

### CODE REVIEW

**Review by Josh-Cena — APPROVED** [review]
clippy-ship-it

File: yarn.lock [hunk 1]
```diff
@@ -2994,15 +2994,14 @@
...
+"@slorber/static-site-generator-webpack-plugin@^4.0.6":
+  version "4.0.6"
+  resolved "https://registry.yarnpkg.com/@slorber/static-site-generator-webpack-plugin/-/static-site-generator-webpack-plugin-4.0.6.tgz#fd460d449800e1ea6e917bf683683db8c8a230bb"
+  integrity sha512-p1EcMCK4Y/y6+O0kAa1HJu7wxx6mHJcOv6z6nFi35CAnO5Krz0fBgfDUAJnmN2Xh1pw/+LCDsN9paKQ5Hou7zA==
   dependencies:
-    bluebird "^3.7.1"
-    cheerio "^0.22.0"
-    eval "^0.1.8"
-    webpack-sources "^1.4.3"
+    eval "^0.1.4"
```
Comment (Josh-Cena) [inline yarn.lock:3002]: This range change looks a bit weird, though

Comment (slorber) [reply yarn.lock:3002]: yes good catch, bad merge, will publish another version

**Comment by Josh-Cena** [comment]
Excited that we got rid of a bunch of lodash modules, bluebird, AND cheerio!

**Pull request merged**